### DISCUSSION TITLE
commit type when dependencies are mixed

### DISCUSSION THREAD

**Discussion by jdbruijn** [renovatebot/renovate#8875] [Request Help] [answered]
I have a situation where I have grouped dependencies in both my `dependencies` and `devDependencies`. I'm using the base configuration, so it is including `:semanticPrefixFixDepsChoreOthers`, which normally works just file. When I have both `dependencies` and `devDependencies, it was using `chore` instead of `fix`, what I was expecting as IMO `fix` type would be superseding `chore` type for the dependency updates.

I have a default config in https://github.com/vidavidorra/.github/blob/main/default.json, and my config that created the PR in question (vidavidorra/commitlint-config#168) is shown below.

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["github>vidavidorra/.github"],
  "packageRules": [
    {
      "description": "Disable updates for aliased @commitlint packages with a final version",
      "packagePatterns": ["^@commitlint/(lint|load)\\-(9|10)\\.x"],
      "depTypeList": ["devDependencies"],
      "enabled": false
    }
  ]
}
```

So basically I was expecting `fix(deps): update commitlint monorepo to v12 (major)` instead of `chore(deps): update commitlint monorepo to v12 (major)`. Is this behaviour expected or do I have something wrong in my configuration?
I know I can also create two groups, like shown below, but I'd like to keep these updates in a single PR due to the testing being linked to the dependencies.

```json

{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["github>vidavidorra/.github"],
  "packageRules": [
    {
      "description": "Disable updates for aliased @commitlint packages with a final version",
      "packagePatterns": ["^@commitlint/(lint|load)\\-(9|10)\\.x"],
      "depTypeList": ["devDependencies"],
      "enabled": false
    },
    {
      "description": "Group commitlint develop dependencies",
      "packagePatterns": ["^@commitlint/"],
      "depTypeList": ["devDependencies"],
      "groupName": "devDependencies commitlint monorepo"
    },
    {
      "description": "Group commitlint dependencies",
      "packagePatterns": ["^@commitlint/"],
      "depTypeList": ["dependencies"],
      "groupName": "dependencies commitlint monorepo"
    }
  ]
}
```

**Comment by rarkins** [comment, marked as answer]
Renovate doesn't understand that fix is higher or more important than chore. The branch will use the prefix of the first update in the branch. 

If you intentionally mix deps and dev deps in groups then it's not compatible with this rule, so you should consider chore only or fix only.

**Reply by jdbruijn** [reply]
I don't intentionally mix deps and dev deps in the Renovate configuration, if I didn't mess that up that is. If you mean in the `package.json` then you're absolutely right as that looks like below.

```json
  "dependencies": {
    "@commitlint/config-conventional": "11.0.0",
    "@commitlint/types": "11.0.0",
    "commitlint-plugin-function-rules": "1.1.23"
  },
  "devDependencies": {
    "@commitlint/cli": "11.0.0",
    "@commitlint/lint-10.x": "npm:@commitlint/lint@10.0.0",
    "@commitlint/lint-11.x": "npm:@commitlint/lint@11.0.0",
    ...
```

So do I understand correctly that I can't create a rule that uses type `chore` if the updates are only dev deps, `fix` if the updates are only deps or deps and dev deps? If so then I think my options are:

1. Split deps and dev deps with the grouping rules in my OP.
2. Always use `fix` for `@commitlint` updates, regardless of it being deps, dev deps or a combination.

**Reply by viceice** [reply]
I think you should always use a group for those, to do not get any dependency conflicts

**Reply by rarkins** [reply]
I'd recommend (2) in that case - keep one group but add an extra rule to force `fix` semantic commit type
